common/SC_Apple.mm

```diff
@@ -41,5 +41,48 @@ void disableAppNap() {
     }
 }
 
+namespace EventLoop {
+
+void setup() {
+#if 0
+    // The following code would transform the process into a foreground application.
+    // For now it's the plugin's responsibility to do this (early or lazily)
+    // because we don't want to always show an icon in the docker.
+    ProcessSerialNumber psn = { 0, kCurrentProcess };
+    TransformProcessType(&psn, kProcessTransformToForegroundApplication);
+#endif
+    // Create NSApplication
+    [NSApplication sharedApplication];
+}
+
+void run() {
+#if 0
+    // this doesn't work...
+    [NSApp run];
+#else
+    // Kudos to https://www.cocoawithlove.com/2009/01/demystifying-nsapplication-by.html
+    NSAutoreleasePool* pool = [[NSAutoreleasePool alloc] init];
+
+    [NSApp finishLaunching];
+    while (true) {
```

Comment:
in the link in the code above, this loop waiting on a condition to be set from `quit`. i think you should do that and let the application exit the same way it used to.

Reply:
The original code actually doesn't work, because most of the time the event loop will be waiting for the next event, so it doesn't have a chance to check the condition. But I can solve that by sending a dummy event.

Reply:
got it! that would probably be best. thanks for figuring out the solution to this, definitely would have taken me a long time